spring-boot-starter/starter-client/spring/src/main/java/org/camunda/bpm/client/spring/impl/client/ClientFactory.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.client.spring.impl.client;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.camunda.bpm.client.ExternalTaskClient;
+import org.camunda.bpm.client.ExternalTaskClientBuilder;
+import org.camunda.bpm.client.backoff.BackoffStrategy;
+import org.camunda.bpm.client.interceptor.ClientRequestInterceptor;
+
+import org.camunda.bpm.client.spring.impl.client.util.ClientLoggerUtil;
+import org.springframework.beans.factory.FactoryBean;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.support.PropertySourcesPlaceholderConfigurer;
+import org.springframework.core.env.PropertyResolver;
+import org.springframework.core.env.PropertySources;
+import org.springframework.core.env.PropertySourcesPropertyResolver;
+import org.springframework.util.CollectionUtils;
+
+@SuppressWarnings("SpringJavaAutowiredMembersInspection")
+public class ClientFactory
+    implements FactoryBean<ExternalTaskClient>, InitializingBean {
+
+  protected static final ClientLoggerUtil LOG = ClientLoggerUtil.CLIENT_LOGGER;
+
+  protected ClientConfiguration clientConfiguration;
+
+  protected BackoffStrategy backoffStrategy;
+  protected List<ClientRequestInterceptor> requestInterceptors = new ArrayList<>();
+
+  protected ExternalTaskClient client;
+
+  protected PropertyResolver propertyResolver;
+
+  @Override
+  public ExternalTaskClient getObject() {
+    if (client == null) {
+      ExternalTaskClientBuilder clientBuilder = ExternalTaskClient.create()
+          .baseUrl(resolve(clientConfiguration.getBaseUrl()))
+          .workerId(resolve(clientConfiguration.getWorkerId()));
```

Comment:
Let's also skip calling these two builder methods when not configured.

Reply:
I agree 👍